models/transaction_test.go

```diff
@@ -397,3 +408,344 @@ func TestValidateTransaction(t *testing.T) {
 	}
 
 }
+
+func TestValidateConsensusRules(t *testing.T) {
```

Comment:
this feels to me like it's testing `txpool.ValidateTransaction` which is test in go-ethereum so I don't feel like we should add tests here

Reply:
Unfortunately the test cases on go-ethereum were very little. And because that method is a rather black-box for us, I wanted to make sure that we validate transaction properly, without false positives / false negatives.

Reply:
I don't understand, aren't all this test existing in go-ethereum already? is there any bellow you added yourself? If not, I don't understand why would we want to have them here.

Reply:
I feel the same, I think this test for Ethereum, we don't need to test this logic.

Reply:
@sideninja @ramtinms These tests do not exist in the `go-ethereum` repository.
The rationale behind them is not to test the logic of `txpool.ValidateTransaction(tx, head, signer, opts)` .
I have added these tests mainly as a safeguard, against future upgrades of the `go-ethereum` package.
And the reason is that this method accepts several parameters.
Suppose that the method is updated and makes use of extra fields from the `head` or `opts` arguments.
Without the tests, any such changes will go unnoticed, and might result in crashes on production.
```go
head := &types.Header{
	Number:   big.NewInt(20_182_324),
	Time:     uint64(time.Now().Unix()),
	GasLimit: 30_000_000,
}
```
For example, right now we only populate the above fields for the `head` argument.